react/usefocus_en.md

```diff
@@ -1,4 +1,44 @@
+There are 3 important points should be aware of:
+1. Reset the state when the target element changes.
+2. Do `addEventListener` & `removeEventListener` **only** when the target element changes, **NOT** on every render.
+3. Keep the handler function as the **same reference** for `addEventListener` & `removeEventListener`, otherwise the handler will **NOT** be removed. There're two mostly used options:
+  - Use `useCallback` to cache the **handler**.
+  - Define the **handler** inside the `useEffect` function scope.(recommended)
 
-There is no solution yet.
+```ts
+import React, { Ref, useEffect, useRef, useState } from 'react'
+
+export function useFocus<T extends HTMLElement>(): [Ref<T | undefined>, boolean] {
+  const [isFocused, setFocused] = useState(false)
+  const ref = useRef<T>()
+
+  useEffect(() => {
+    const currentElement = ref.current
+
+    // !!IMPORTANT!! 
+    // reset the focus state when the currentElement changes.
+    setFocused(document.activeElement === currentElement)
```

Comment:
why  is this necessary ? the hook check the focus state doesn't necessarily mean we need to manually set the focus.

Reply:
It's a initinalization when `currentElement(ref.current)` changes, there might be dirty value if without this initinalization.

**Let me try to explain this:**
1. At the begining, the `currentElement` is `target0` and it's focused, then the `focus` state goes `true`.
2. Change `currentElement` from `target0` to `target1`(no `blur` fired on `target0`), now the state will still be `true`, while the new `target1` is not focused. 

try it online:
https://codesandbox.io/s/exciting-mendeleev-lmz5iu?file=/src/useFocus.js

Reply:
Great catch, guess we need to update the test cases to cover this.
Wondering if we should do this in useEffect cleanup function?